agent/hcp/telemetry/otel_sink_test.go

```diff
@@ -0,0 +1,386 @@
+
+	sink.IncrCounter([]string{"consul", "raft", "state", "leader"}, float32(23.23))
+	sink.IncrCounterWithLabels([]string{"consul", "raft", "apply"}, float32(1.44), labels)
+
+	sink.AddSample([]string{"consul", "raft", "leader", "lastContact"}, float32(45.32))
+	sink.AddSampleWithLabels([]string{"consul", "raft", "commitTime"}, float32(26.34), labels)
+
+	var collected metricdata.ResourceMetrics
+	err = reader.Collect(ctx, &collected)
+	require.NoError(t, err)
+
+	isSame(t, expectedSinkMetrics, collected)
+}
+
+func TestOTELSink_Race(t *testing.T) {
+	reader := metric.NewManualReader()
+	ctx := context.Background()
+	opts := &OTELSinkOpts{
+		Ctx:    ctx,
+		Reader: reader,
+	}
+
+	sink, err := NewOTELSink(opts)
+	require.NoError(t, err)
+
+	samples := 100
+	expectedMetrics := generateSamples(samples)
+	wg := &sync.WaitGroup{}
+	errCh := make(chan error, samples)
+	for k, v := range expectedMetrics {
+		wg.Add(1)
+		go func(k string, v metricdata.Metrics) {
+			performSinkOperation(t, sink, k, v, errCh)
+			wg.Done()
+		}(k, v)
+	}
+	wg.Wait()
+
+	require.Empty(t, errCh)
+
+	var collected metricdata.ResourceMetrics
+	err = reader.Collect(ctx, &collected)
+	require.NoError(t, err)
+
+	isSame(t, expectedMetrics, collected)
+}
+
+// generateSamples generates n of each gauges, counter and histogram measurements to use for test purposes.
+func generateSamples(n int) map[string]metricdata.Metrics {
+	generated := make(map[string]metricdata.Metrics, 3*n)
+
+	for i := 0; i < n; i++ {
+		v := 12.3
+		k := fmt.Sprintf("consul.test.gauges.%d", i)
+		generated[k] = metricdata.Metrics{
+			Name: k,
+			Data: metricdata.Gauge[float64]{
+				DataPoints: []metricdata.DataPoint[float64]{
+					{
+						Attributes: *attribute.EmptySet(),
+						Value:      float64(float32(v)),
+					},
+				},
+			},
+		}
+	}
+
+	for i := 0; i < n; i++ {
+		v := 22.23
+		k := fmt.Sprintf("consul.test.sum.%d", i)
+		generated[k] = metricdata.Metrics{
+			Name: k,
+			Data: metricdata.Sum[float64]{
+				DataPoints: []metricdata.DataPoint[float64]{
+					{
+						Attributes: *attribute.EmptySet(),
+						Value:      float64(float32(v)),
+					},
+				},
+			},
+		}
+
+	}
+
+	for i := 0; i < n; i++ {
+		v := 13.24
+		k := fmt.Sprintf("consul.test.hist.%d", i)
+		generated[k] = metricdata.Metrics{
+			Name: k,
+			Data: metricdata.Histogram[float64]{
+				DataPoints: []metricdata.HistogramDataPoint[float64]{
+					{
+						Attributes: *attribute.EmptySet(),
+						Sum:        float64(float32(v)),
+						Max:        metricdata.NewExtrema(float64(float32(v))),
+						Min:        metricdata.NewExtrema(float64(float32(v))),
+						Count:      1,
+					},
+				},
+			},
+		}
+	}
+
+	return generated
+}
+
+// performSinkOperation emits a measurement using the OTELSink and calls wg.Done() when completed.
+func performSinkOperation(t *testing.T, sink *OTELSink, k string, v metricdata.Metrics, errCh chan error) {
```

Comment:
I don't think we need `t` anymore? 
```suggestion
func performSinkOperation(sink *OTELSink, k string, v metricdata.Metrics, errCh chan error) {
```